### PR TITLE
[Distributed] Fix issues detected by ASAN

### PR DESCRIPTION
**Pull request by ktoso**
PR to resolve silly mistakes which actually were unsafe, thank you c++ 👋 😢 

resolves rdar://88301320

Also unlocking all tests again to see if we good again here rdar://88228867

### CODE REVIEW

**Comment by ktoso** [comment]
The first commit here is https://github.com/apple/swift/pull/41154 which will be meged in a moment I hope...

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
@swift-ci asan test

**Comment by ktoso** [comment]
Resolves rdar://88301320

**Comment by ktoso** [comment]
@swift-ci asan test

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Review by xedin — APPROVED** [review]

**Comment by ktoso** [comment]
@swift-ci asan test

**Comment by ktoso** [comment]
@swift-ci please test

File: lib/Sema/CodeSynthesisDistributedActor.cpp [hunk 1]
```diff
@@ -145,10 +145,19 @@ VarDecl *GetDistributedActorSystemPropertyRequest::evaluate(
...
+
+  for (auto system : actor->lookupDirect(C.Id_actorSystem)) {
+    if (auto var = dyn_cast<VarDecl>(system)) {
+      auto conformance = module->conformsToProtocol(var->getInterfaceType(),
+                                                    DistSystemProtocol);
+      if (conformance.isInvalid())
+        continue;
+
+      return var;
+    }
```
Comment (ktoso) [inline lib/Sema/CodeSynthesisDistributedActor.cpp:160]: "the fix"™

**Comment by ktoso** [comment]
Linux failure was unrelated `08:05:39 b"Cloning into 'llvm-project'...\nremote: fatal: packfile /data/repositories/2/nw/2d/85/ea/138117697/network.git/objects/pack/pack-337de53411b5e04963ebdfdf7c9a6ccf249bf504.pack cannot be accessed        \nremote: aborting due to possible repository corruption on the remote side.\nfatal: early EOF\nfatal: index-pack failed\n"`

**Comment by ktoso** [comment]
Tests passed but we want to try to reproduce if it isn't flaky.

**Comment by ktoso** [comment]
@swift-ci please test and merge

**Comment by ktoso** [comment]
@swift-ci please test

**Comment by ktoso** [comment]
@swift-ci please test and merge

**Comment by ktoso** [comment]
resolves rdar://88228867

**Pull request merged**